dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java

```diff
@@ -984,7 +989,23 @@ public void mapAllPermissions() throws DotDataException {
 		try {
 
 			Logger.debug( PermissionAPI.class, String.format("::getRoles -> before loading inode object(%s)", inode) );
-			inodeObj = InodeFactory.getInode(inode, Inode.class);
+			
+			//Using the ShortyAPI to identify the nature of this inode
+            Optional<ShortyId> shortOpt = APILocator.getShortyAPI().getShorty(inode);
+            
+          //Hibernate won't handle structures, thats why we need a special case here
+            if ( ShortType.STRUCTURE == shortOpt.get().subType ) {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "shortOpt.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)